broadcastclient/broadcastclient.go

```diff
@@ -311,11 +327,25 @@ func (bc *BroadcastClient) startBackgroundReader(earlyFrameData io.Reader) {
 					connected = false
 					sourcesConnectedGauge.Dec(1)
 					sourcesDisconnectedGauge.Inc(1)
+					if sourcesConnectedGauge.Value() <= 0 {
+						log.Error("feed server was just disconnected")
+					}
 				}
 				_ = bc.conn.Close()
+				timer := time.NewTimer(backoffDuration)
+				if backoffDuration < bc.config.ReconnectMaximumBackoff {
+					backoffDuration *= 2
+				}
+				select {
+				case <-ctx.Done():
+					timer.Stop()
+					return
+				case <-timer.C:
+				}
 				earlyFrameData = bc.retryConnect(ctx)
```

Comment:
It seems to me like the backoff implemented in this function should be used instead of the timer outside the function

Reply:
If tcp connection is able to reconnect easily, but websocket initialization fails, there will be rapidly repeating error messages.  This has happened often in production.